### ISSUE TITLE
Test framework fails on OSX

### ISSUE THREAD

**Issue by DanielO** [servalproject/serval-dna#4]
OSX Mountain Lion has old Bash (3.2.48) and tests fail like so..

```

[ur 11:43] ~/projects/serval/batphone/jni/serval-dna >./tests/dnahelper 
1. Non-existent DNA helper executable...2. DNA helper configured argument...3. DNA helper returns one valid reply...4. DNA helper returns two valid replies...5. DNA helper returns three valid replies...6. DNA helper returns empty URI...7. DNA helper returns invalid URI, missing scheme...8. DNA helper returns invalid URI, invalid char...9. DNA helper returns invalid URI, empty hierarchical part...10. DNA helper returns mismatched token...11. DNA helper returns empty token...12. DNA helper returns invalid token...13. DNA helper returns invalid token, too long...14. DNA helper returns invalid token, too short...15. DNA helper returns mismatched DID...16. DNA helper returns empty DID...17. DNA helper returns invalid DID...18. DNA helper returns invalid DID, too long...19. DNA helper returns invalid DID, too short...20. DNA helper returns invalid reply, missing delimiter...21. DNA helper returns invalid reply, name too long...22. DNA helper returns invalid reply, empty line...23. DNA helper returns invalid reply, missing newline...24. DNA helper process takes too long to reply and is restarted...25. DNA helper spurious output after DONE is ignored...26. DNA helper process dies unexpectedly and is restarted...26 tests, 0 pass, 0 fail, 0 error
[ur 11:45] ~/projects/serval/batphone/jni/serval-dna >./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ExecArg1: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ExecError: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/TokenInvalidLong: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/DidMismatch: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/UriInvalid2: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/UriInvalid3: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/TokenInvalid: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/TokenInvalidShort: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ReplyOk2: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/UriEmpty: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/UriInvalid1: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/DidInvalid: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ReplyOk3: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/TokenMismatch: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/DidInvalidLong: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ReplyInvalidLongName: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ReplyOk1: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/DidEmpty: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/TokenEmpty: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ReplyInvalidMissingDelim: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/DidInvalidShort: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ReplyInvalidMissingNewline: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ReplyInvalidEmpty: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/HelperDies: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/ReplySpurious: No such file or directory
./tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-28333/results/HelperTimeout: No such file or directory
```

**Comment by DanielO** [comment]
I tried installing a newer bash via MacPorts but no change.

```

[ur 11:51] ~/projects/serval/batphone/jni/serval-dna >/opt/local/bin/bash --version      
GNU bash, version 4.2.37(2)-release (i386-apple-darwin12.0.0)
Copyright (C) 2011 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
[ur 11:51] ~/projects/serval/batphone/jni/serval-dna >tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ExecArg1: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ExecError: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/UriInvalid1: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ReplyOk3: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/TokenMismatch: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ReplyOk2: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ReplyOk1: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/UriInvalid3: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/UriInvalid2: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/UriEmpty: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/TokenEmpty: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/DidEmpty: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/DidMismatch: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/DidInvalid: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/DidInvalidShort: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/TokenInvalid: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/TokenInvalidShort: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/TokenInvalidLong: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ReplyInvalidLongName: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ReplyInvalidMissingDelim: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ReplyInvalidEmpty: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/DidInvalidLong: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ReplyInvalidMissingNewline: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/HelperDies: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/ReplySpurious: No such file or directory
tests/../testframework.sh: line 239: /var/folders/vw/9v3c0ztd1qj3ry2b_2cbrqfh0000gn/T//_tfw-57901/results/HelperTimeout: No such file or directory
```

**Comment by DanielO** [comment]
The last version of testframework.sh where this works is e1fb64e9d458cce8cc8aa26b08558eaca34298ba

**Comment by quixotique** [comment]
Fixed by 3a6b424c2daa376ec219bd2167782f4c445f659c
